core/sds-aspect-model-resolver/src/main/java/io/openmanufacturing/sds/aspectmodel/resolver/ClasspathStrategy.java

```diff
@@ -88,18 +97,59 @@ protected URL resourceUrl( final String directory, final String filename ) {
 
    protected Stream<String> filesInDirectory( final String directory ) {
       try {
-         final InputStream directoryUrl = getClass().getResourceAsStream( directory );
-         if ( directoryUrl == null ) {
-            LOG.warn( "No such classpath directory {}", directory );
-            return Stream.empty();
+         final Optional<File> file = getDirectoryFile(directory);
+         if(file.isPresent() && file.get().isFile()) {
+            return getFilesFromJar( directory, file.get() );
+         } else {
+            return getFilesFromResources( directory );
          }
-         return Arrays.stream( IOUtils.toString( directoryUrl, StandardCharsets.UTF_8 ).split( "\\n" ) );
       } catch ( final IOException exception ) {
          LOG.warn( "Could not list files in classpath directory {}", directory, exception );
          return Stream.empty();
       }
    }
 
+   private Optional<File> getDirectoryFile( String directory ) {
+      final URL url = getClass().getClassLoader().getResource( directory );
+      if(url == null)
+      {
+         return Optional.empty();
+      }
+      final String urlString = url.toString();
```

Comment:
Instead of this logic, couldn't you just use `new File(url.toURI())`? Probably not because of the "jar:file:", but then please add a comment on the method.

Reply:
Added a comment to it as i've seen no other way to get the information needed.